File: app/src/main/java/com/homehijack/homehijacking/MainActivity.java
package com.homehijack.homehijacking;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {
    private BottomNavigationView bottomNavigationView;
    private ReservationFragment reservationFragment;
    private FriendFragment friendFragment;
    private SettingFragment settingFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        bottomNavigationView = (BottomNavigationView) findViewById(R.id.main_bnv);

        reservationFragment = new ReservationFragment();
        friendFragment = new FriendFragment();
        settingFragment = new SettingFragment();

        getSupportFragmentManager().beginTransaction().replace(R.id.main_fl, reservationFragment).commitAllowingStateLoss();

        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                switch (item.getItemId()) {
                    case R.id.reservation: {
                        getSupportFragmentManager().beginTransaction().replace(R.id.main_fl, reservationFragment).commitAllowingStateLoss();
                        return true;
                    }
                    case R.id.friend: {
                        getSupportFragmentManager().beginTransaction().replace(R.id.main_fl, friendFragment).commitAllowingStateLoss();
                        return true;
                    }
                    case R.id.setting: {
                        getSupportFragmentManager().beginTransaction().replace(R.id.main_fl, settingFragment).commitAllowingStateLoss();
                        return true;
                    }
                    default:
                        return false;
                }
            }
        });

    }
}

File: app/src/main/java/com/homehijack/homehijacking/MainActivity.kt
package com.homehijack.homehijacking

import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemSelectedListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val bottomNavigationView = findViewById<View>(R.id.bottomNavigationView) as BottomNavigationView




    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when(item.itemId){
            R.id.pageReservation ->{
                supportFragmentManager.beginTransaction().replace(R.id.mainLayout, ReservationFragment()).commitAllowingStateLoss()
                return true
            }
            R.id.pageFriends ->{
                supportFragmentManager.beginTransaction().replace(R.id.mainLayout, FriendsFragment()).commitAllowingStateLoss()
                return true
            }
            R.id.pageSettings ->{
                supportFragmentManager.beginTransaction().replace(R.id.mainLayout, SettingsFragment()).commitAllowingStateLoss()
                return true
            }
        }
        return false
    }

}